### PR TITLE
Link the memory viewer with the debugger

### PR DESCRIPTION
**Pull request by elad335**
* Press M while the debugger is running to launch the memory viewer from it. When the memoy viewer is being launched from the debugger a magical thing happens: it passes to it the address of current instruction selected on the debugger. (or the current PC if none was selected)
It also works with SPUs and the debugger calculates the address of the LS shadow copy in vm and adds it to the address before then passing it to the memory viewer.

### CODE REVIEW

File: rpcs3/rpcs3qt/memory_viewer_panel.cpp [hunk 1]
```diff
@@ -46,7 +46,7 @@ memory_viewer_panel::memory_viewer_panel(QWidget* parent)
 	QGroupBox* tools_mem_addr = new QGroupBox(tr("Address"));
 	QHBoxLayout* hbox_tools_mem_addr = new QHBoxLayout();
 	m_addr_line = new QLineEdit(this);
-	m_addr_line->setPlaceholderText("00000000");
+	m_addr_line->setText(qstr(fmt::format("%x", m_addr)));
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/memory_viewer_panel.cpp:49]: Why remove the placeholder text. It has nothing to do with the text

**Comment by Megamouse** [comment]
This probably allows you to show multiple viewers. Is this intended?

**Comment by elad335** [comment]
I don't it's a bad thing to have multiple viewer espacially if they are for different memory sections entirely.
I just tried and master and it also allows multiple viewers.

File: rpcs3/rpcs3qt/memory_viewer_panel.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/rpcs3qt/memory_viewer_panel.cpp]: Fixed.

**Comment by Megamouse** [comment]
why are there 2 PRs with the same commit ?

**Comment by elad335** [comment]
Fixed.

File: rpcs3/rpcs3qt/memory_viewer_panel.cpp [hunk 2]
```diff
@@ -46,7 +46,7 @@ memory_viewer_panel::memory_viewer_panel(QWidget* parent)
 	QGroupBox* tools_mem_addr = new QGroupBox(tr("Address"));
 	QHBoxLayout* hbox_tools_mem_addr = new QHBoxLayout();
 	m_addr_line = new QLineEdit(this);
-	m_addr_line->setPlaceholderText("00000000");
+	m_addr_line->setPlaceholderText(qstr(fmt::format("%08x", m_addr)));
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/memory_viewer_panel.cpp:49]: what I meant was: keep both.

Comment (elad335) [reply rpcs3/rpcs3qt/memory_viewer_panel.cpp]: Fixed.

**Pull request merged**